### PR TITLE
[Doc] Define the user roles

### LINKED ISSUE #192: Describe the user roles more comprehensively
We should describe:

- how many roles there are
- what each can do
- what are their responsibility

### PR DESCRIPTION
**Pull request by Huy-Ngo**
Expectedly resolve #192

The rendered table looks ugly due to word breaking tho.

### CODE REVIEW

**Comment by Huy-Ngo** [comment]
Hmm I think I should add the inheritance model here too.

**Comment by Huy-Ngo** [comment]
Eh... that model is not implemented anyway... should we include that?

File: docs/source/req/users.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Student     progress the project       manage projects,               
+                                       manage and complete tasks,     
+                                       upload reports and slides      
+Supervisor  support the group and      create tasks, evaluate projects
+            evaluate the project                                      
+Judge       evaluate groups'           access and evaluate            
+            reports and presentations  reports and presentation       
+Assistant   collect statistical        aggregate and export           
+            report and evaluation      projects' evaluation           
+==========  =========================  ===============================
```
Comment (McSinyx) [inline docs/source/req/users.rst:45]: > The rendered table looks ugly due to word breaking tho.

See https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-tabularcolumns

**Comment by McSinyx** [comment]
As Raymond Hettinger once said, inheritance is for code reuse, and we do reuse code handling students and supervisors (in other words, there are places where they're treated the same).  What you wrote described that well enough so I'm not sure what to add.

**Comment by Huy-Ngo** [comment]
>  What you wrote described that well enough so I'm not sure what you planned to add.

I planned to add the UML for it, i.e. this one

UML Diagram

By the way, since jury and supervisor can be the same person, what are we gonna do about it?

**Comment by McSinyx** [comment]
> since judge and supervisor can be the same person (though for different projects), what are we gonna do about it?

I'd suggest adding a field *judges* to *project* which is differently invited (i.e. separate *invite member* from *invite judge*).

**Comment by Huy-Ngo** [comment]
I think judge can be added by the faculty's assistant.

**Comment by Huy-Ngo** [comment]
Referenced in #197

**Review by McSinyx — APPROVED** [review]

**Pull request merged**